include/deal.II/lac/solver_cg.h

```diff
@@ -389,6 +462,481 @@ SolverCG<VectorType>::compute_eigs_and_cond(
+                                                         length,
+                                                         r,
+                                                         prec_r.data());
+                  DEAL_II_OPENMP_SIMD_PRAGMA
+                  for (unsigned int i = 0; i < length; ++i)
+                    {
+                      p[i] = this->beta * p[i] + prec_r[i];
+                      v[i] = Number();
+                    }
+                  p += length;
+                  r += length;
+                  v += length;
+                }
+            }
+          else
+            {
+              const Number alpha_plus_previous_alpha_over_beta =
+                this->alpha + this->previous_alpha / this->previous_beta;
+              const Number previous_alpha_over_beta =
+                this->previous_alpha / this->previous_beta;
+              for (unsigned int j = start_range; j < end_range; j += grain_size)
+                {
+                  const unsigned int length =
+                    std::min(grain_size, end_range - j);
+                  this->preconditioner.apply_to_subrange(j,
+                                                         length,
+                                                         r,
+                                                         prec_r.data());
+                  DEAL_II_OPENMP_SIMD_PRAGMA
+                  for (unsigned int i = 0; i < length; ++i)
+                    {
+                      x[i] += alpha_plus_previous_alpha_over_beta * p[i] -
+                              previous_alpha_over_beta * prec_r[i];
+                      r[i] -= this->alpha * v[i];
+                    }
+                  this->preconditioner.apply_to_subrange(j,
+                                                         length,
+                                                         r,
+                                                         prec_r.data());
+                  DEAL_II_OPENMP_SIMD_PRAGMA
+                  for (unsigned int i = 0; i < length; ++i)
+                    {
+                      p[i] = this->beta * p[i] + prec_r[i];
+                      v[i] = Number();
+                    }
+                  p += length;
+                  r += length;
+                  v += length;
+                  x += length;
+                }
+            }
+        };
+
+        std::array<Number, 7> local_sums = {};
+        const auto operation_after_loop  = [&](const unsigned int start_range,
+                                              const unsigned int end_range) {
+          const Number *                 x = this->x.begin() + start_range;
+          const Number *                 r = this->r.begin() + start_range;
+          const Number *                 p = this->p.begin() + start_range;
+          const Number *                 v = this->v.begin() + start_range;
+          std::array<Number, grain_size> prec_r;
+          std::array<Number, grain_size> prec_v;
```

Comment:
This is new. We need to cache the intermediate result; it also means that `grain_size` must be a compile constant...

Reply:
I remember I have seen the same issues 2 years ago. Do you have a feeling how much/if any penalty is introduced by this?